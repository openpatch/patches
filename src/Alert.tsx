import { ReactNode } from "react";
import { Box, BoxProps } from "./Box";
import { Icon } from "./Icon";
import { Check, Important, Information } from "./icons/shade";

export type AlertProps = {
  children?: ReactNode;
  severity?: "neutral" | "warning" | "error" | "success" | "info";
};

export const Alert = ({ children, severity }: AlertProps) => {
  let backgroundColor: BoxProps["backgroundColor"] = "neutral.50";
  let textColor: BoxProps["textColor"] = "neutral.900";
  let icon: ReactNode = <Information />;
  switch (severity) {
    case "info":
      backgroundColor = "info.50";
      textColor = "info.900";
      icon = <Information />;
      break;
    case "error":
      backgroundColor = "error.50";
      textColor = "error.900";
      icon = <Important />;
      break;
    case "success":
      backgroundColor = "success.50";
      textColor = "success.900";
      icon = <Check />;
      break;
    case "warning":
      backgroundColor = "warning.50";
      textColor = "warning.900";
      icon = <Important />;
      break;
  }
  return (
    <Box
      display="flex"
      backgroundColor={backgroundColor}
      textColor={textColor}
      boxShadow="standard"
      padding="standard"
      borderColor={textColor}
      borderStyle="solid"
      borderWidth="standard"
      borderRadius="standard"
    >
      <Icon size="small" color={severity}>
        {icon}
      </Icon>
      <Box marginLeft="xsmall" flex="1">
        {children}
      </Box>
    </Box>
  );
};
