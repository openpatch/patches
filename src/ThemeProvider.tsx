import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
