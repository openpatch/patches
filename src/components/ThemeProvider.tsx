import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
