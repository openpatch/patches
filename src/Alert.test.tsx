import { render } from "@testing-library/react";
import { Alert, AlertProps } from "./Alert";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: AlertProps) =>
  render(
    <ThemeProvider>
      <Alert {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
