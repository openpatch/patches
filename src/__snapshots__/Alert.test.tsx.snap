// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #F5F7FA;
  color: #1F2933;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1),0 1px 2px 0 rgba(0, 0, 0, 0.06);
  padding: 16px;
  border-color: #1F2933;
  border-style: solid;
  border-width: 1px;
  border-radius: 8px;
}

.emotion-1 {
  position: relative;
  border-radius: 9999px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #52606D;
  width: 24px;
  height: 24px;
}

.emotion-1 svg {
  width: 24px;
  height: 24px;
}

.emotion-1 svg .primary {
  fill: #9AA5B1;
}

.emotion-1 svg .secondary {
  fill: #52606D;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  margin-left: 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div
    id="patches-root"
  >
    <div
      class="emotion-0"
      display="flex"
    >
      <span
        class="emotion-1"
      >
        <svg
          class="icon-information"
          viewBox="0 0 24 24"
        >
          <path
            class="primary"
            d="M12 2a10 10 0 110 20 10 10 0 010-20z"
          />
          <path
            class="secondary"
            d="M11 12a1 1 0 010-2h2a1 1 0 01.96 1.27L12.33 17H13a1 1 0 010 2h-2a1 1 0 01-.96-1.27L11.67 12H11zm2-4a1.5 1.5 0 110-3 1.5 1.5 0 010 3z"
          />
        </svg>
      </span>
      <div
        class="emotion-2"
      />
    </div>
  </div>
</div>
`;
